test/benchmark-common-tasks/main.cpp

```diff
@@ -593,6 +596,95 @@ struct BenchmarkQueryLongString : BenchmarkWithStrings {
     }
 };
 
+struct BenchmarkQueryInsensitiveString : BenchmarkWithStringsTable {
+    const char* name() const
+    {
+        return "QueryInsensitiveString";
+    }
+
+    std::string gen_random_case_string(size_t length)
+    {
+        std::stringstream ss;
+        for (size_t c = 0; c < length; ++c) {
+            bool lowercase = (rand() % 2) == 0;
+            // choose characters from a-z or A-Z
+            ss << (rand() % 26) + (lowercase ? 97 : 65);
+        }
+        return ss.str();
+    }
+
+    std::string shuffle_case(std::string input)
+    {
+        for (size_t i = 0; i < input.size(); ++i) {
+            bool upper = (rand() % 2) == 0;
+            int c = input[i];
```

Comment:
`input[i]` returns a `char` and `input[i] = c` yields a compiler warning if `c` is not of type `char`.